source/robot_model/src/Model.cpp

```diff
@@ -330,4 +331,15 @@ void Model::print_qp_problem() {
     std::cout << this->upper_bound_constraints_(i) << std::endl;
   }
 }
+
+bool Model::create_urdf_from_string(const std::string& urdf_string, const std::string& desired_path) {
+  std::ofstream file(desired_path);
+  if (!file.good() || !file.is_open()) {
+    return false;
+  } else {
+    file << urdf_string;
+    file.close();
+    return true;
+  }
```

Comment:
This can be simplified a little bit
```c++
if (file.good() && file.is_open()) {
  file << urdf_string;
  file.close();
  return true;
}
return false;
```

Reply:
One day I'm gonna have it right the first time :smile:

Reply:
The rule of thumb is: avoid an `else` branch when you can. Sometimes that means checking for failure first, and sometimes it means checking for success first. It always depends on context!